htsohm/htsohm.py

```diff
@@ -13,10 +13,22 @@
 
 import htsohm
 from htsohm import config
-from htsohm.db import engine, session, Material, MutationStrength
-from htsohm.material_files import generate_pseudo_material, mutate_pseudo_material
+from htsohm.db import engine, session, Material, MutationStrength, Structure
+from htsohm.material_files import generate_material, mutate_material
 from htsohm import simulation
 
+def clone_material(material):
+    copy = material.clone()
+    if material.structure:
+        copy.structure = material.structure.clone()
+        if material.structure.atom_sites:
+            for atom_site in material.structure.atom_sites:
+                copy.structure.atom_sites.append(atom_site)
+        if material.structure.lennard_jones:
+            for atom_type in material.structure.lennard_jones:
+                copy.structure.lennard_jones.append(atom_type)
+    return copy
+
```

Comment:
1. override clone() on your Structure class to handle atom_sites and lennard_jones. It can call the existing clone to start out.
2. override clone() on your Material class. It can use the existing clone as a start. It can call the existing clone to start out.

Come talk to me before you attempt this to make sure we are on the same page.

Also consider using python's lib copy to refactor the Material.clone() method... Might simplify that code?